src/containers/UserEdit.jsx

```diff
@@ -5,91 +5,285 @@ import wrapMutations from './hoc/wrap-mutations'
 import gql from 'graphql-tag'
 
 import GSForm from '../components/forms/GSForm'
-import GSSubmitButton from '../components/forms/GSSubmitButton'
 import Form from 'react-formal'
 import yup from 'yup'
-
-import FlatButton from 'material-ui/FlatButton'
+import Dialog from 'material-ui/Dialog'
 import RaisedButton from 'material-ui/RaisedButton'
+import { StyleSheet, css } from 'aphrodite'
+
 import { dataTest } from '../lib/attributes'
 
-class UserEdit extends React.Component {
+const styles = StyleSheet.create({
+  buttons: {
+    display: 'flex'
+  },
+  container: {
+    display: 'inline-block',
+    marginRight: 20,
+    marginTop: 15
+  }
+})
 
+class UserEdit extends React.Component {
   constructor(props) {
     super(props)
+    this.state = {
+      changePasswordDialog: false,
+      successDialog: false
+    }
     this.handleSave = this.handleSave.bind(this)
+    this.handleClick = this.handleClick.bind(this)
+    this.handleClose = this.handleClose.bind(this)
+    this.openSuccessDialog = this.openSuccessDialog.bind(this)
+    this.buildFormSchema = this.buildFormSchema.bind(this)
   }
 
-  state = {
-    finished: false,
-    stepIndex: 0
-  };
-
   async componentWillMount() {
-    const user = await this.props.mutations.editUser(null)
+    if (!this.props.authType) {
+      await this.props.mutations.editUser(null)
+    }
   }
 
   async handleSave(formData) {
-    const result = await this.props.mutations.editUser(formData)
-    if (this.props.onRequestClose) {
-      this.props.onRequestClose()
+    if (!this.props.authType) {
+      await this.props.mutations.editUser(formData)
+      if (this.props.onRequestClose) {
+        this.props.onRequestClose()
+      }
+    } if (this.props.authType === 'change') {
```

Comment:
is this an `else if`?  otherwise it should be a newline here.  It wouldn't matter for this condition-block, but the `else` block below will e.g. trigger for all non-authType=change cases.

Reply:
Good catch! It should be an `else if`.